scripts/merge.cr

```diff
@@ -8,17 +8,22 @@ def level(sponsor : Sponsor)
   LEVELS.find { |amount| amount <= sponsor.last_payment.to_i }.not_nil!
 end
 
-all_sponsors = Array(Sponsor).new
+all_sponsors_map = Hash(UInt64, Sponsor).new
 overrides = Array(Sponsor).new
 
 %w(opencollective.json bountysource.json others.json).each do |filename|
   File.open("#{__DIR__}/../_data/#{filename}") do |file|
     sponsors = Array(Sponsor).from_json(file)
     sponsors, overrides = sponsors.partition(&.overrides.nil?) if filename == "others.json"
-    all_sponsors.concat sponsors
+    sponsors.each do |sponsor|
+      sponsor_hash = sponsor.name.hash ^ (sponsor.url || "").hash # relevant fields to merge on
+      all_sponsors_map[sponsor_hash] = sponsor.merge(all_sponsors_map[sponsor_hash]?)
+    end
   end
 end
 
+all_sponsors = all_sponsors_map.values.flatten # select all sponsors now that they've been merged
```

Comment:
```suggestion
all_sponsors = all_sponsors_map.values # select all sponsors now that they've been merged
```
`flatten` shouldn't be necessary because the array is already flat.